pandas/tests/series/methods/test_shift.py

```diff
@@ -276,3 +276,20 @@ def test_shift_dt64values_int_fill_deprecated(self):
 
         expected = pd.Series([pd.Timestamp(0), ser[0]])
         tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize("periods", [1, 2, 3, 4])
+    def test_shift_preserve_freqstr(self, periods):
+        # GH#21275
+        data = list(range(0, periods))
+        ser = pd.Series(
+            data, index=pd.date_range("2016-1-1 00:00:00", periods=periods, freq="H")
+        )
+
+        result = ser.shift(1, "2H")
+
+        expected = pd.Series(
+            data, index=pd.date_range("2016-1-1 02:00:00", periods=periods, freq="H")
+        )
+        assert str(np.dtype(result)) == "int64"
+        assert result.index.freqstr == "H"
+        tm.assert_series_equal(result, expected)
```

Comment:
I think assert_series_equal should be enough here since it'll check the index frequencies and dtypes